adlfs/spec.py

```diff
@@ -892,7 +898,7 @@ async def _ls(
                                                 outblobs.append(blob_)
                                             elif (
                                                 blob_["name"].count("/") == depth
-                                                and blob_["size"] == 0
+                                                and (hasattr(blob_, "size") and blob_["size"] == 0)
```

Comment:
Can you share a bit about why the `blob_` would have a size attribute? Does the lack of `size` attribute indicate a "directory" marker?

Reply:
Yes Tom, have seen the blob not having size if the blob type is not a file.